src/Service.Tests/SqlTests/RestApiTests/Find/FindApiTestBase.cs

```diff
@@ -1553,7 +1570,26 @@ public async Task FindTestWithFirstZeroSingleKeyPagination()
                 entityNameOrPath: _integrationEntityName,
                 sqlQuery: string.Empty,
                 exceptionExpected: true,
-                expectedErrorMessage: "Invalid number of items requested, $first must be an integer greater than 0. Actual value: 0",
+                expectedErrorMessage: "Invalid number of items requested, $first must be -1 or an integer greater than 0. Actual value: 0",
+                expectedStatusCode: HttpStatusCode.BadRequest
+            );
+        }
+
+        /// <summary>
+        /// Tests the REST Api for Find operation using $first=100001
+        /// to request > max records of 100000 records, which should throw a DataApiBuilder
```

Comment:
nit: specify that this test depends on runtime config default of min/max page size property being set to xyz.

Reply:
fixed.